src/main/java/seedu/address/logic/commands/ListCommand.java

```diff
@@ -1,24 +1,117 @@
 package seedu.address.logic.commands;
 
 import static java.util.Objects.requireNonNull;
-import static seedu.address.model.Model.PREDICATE_SHOW_ALL_STUDYSPOTS;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
 
 import seedu.address.model.Model;
+import seedu.address.model.studyspot.StudySpot;
+import seedu.address.model.tag.Tag;
 
 /**
  * Lists all study spots in the study tracker to the user.
  */
 public class ListCommand extends Command {
 
     public static final String COMMAND_WORD = "list";
+    public static final String FLAG_FAVOURITES = "f";
+    public static final String FLAG_TAGS = "t";
 
+    public static final String MESSAGE_CONSTRAINTS = "Only valid flags are accepted as extra parameters";
```

Comment:
Right now this isn't triggered anywhere in the code. I can have undefined flags like `-abcd` and the ListCommand still gets parsed.